### PR TITLE
Couch to SQL migration fixes and cleanup

### PR DESCRIPTION
**Pull request by millerdev**
Review commits individually. The final commit adds a new feature that tracks migration dates.

@dannyroberts @snopoke cc @Rohit25negi

### CODE REVIEW

**Review by snopoke — APPROVED** [review]

**Comment by dannyroberts** [comment]
Seeing these failures on builds:
```
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 288, in _get_data
    qc = self.query_context(start=start, limit=limit)
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 264, in query_context
    'queries': self.get_sql_queries()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 301, in get_sql_queries
    qc = self.query_context()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 264, in query_context
    'queries': self.get_sql_queries()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 301, in get_sql_queries
    qc = self.query_context()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 264, in query_context
    'queries': self.get_sql_queries()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 301, in get_sql_queries
    qc = self.query_context()
  File "/mnt/commcare-hq-ro/corehq/apps/reports/sqlreport.py", line 264, in query_context
    'queries': self.get_sql_queries()
...
 ```

looks like a mutually-recursive loop somewhere. Not sure if this is related to your changes or not

**Comment by snopoke** [comment]
hmm, that build failure looks like it's my fault, fix coming

**Comment by snopoke** [comment]
Build fix here: https://github.com/dimagi/commcare-hq/pull/23867

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 1]
```diff
@@ -101,6 +101,10 @@ def log_error(self, message):
         _logger.error(message)
         print('[ERROR] {}'.format(message))
 
+    def log_warning(self, message):
+        _logger.warning(message)
+        print('[WARNING] {}'.format(message))
+
```
Comment (kaapstorm) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: These log methods might be getting to the point where you'd want to DRY them by getting them to call a generic `_log()` method like
```python
    def _log(self, level, message):
        _logger.log(level, message)
        if level != logging.DEBUG or self.debug:
            print('[{}] {}'.format(logging.getLevelName(level), message))
```
with, say, 
```python
    def log_debug(self, message):
        self._log(logging.DEBUG, message)
```
(the way [logging](https://github.com/python/cpython/blob/master/Lib/logging/__init__.py#L1433) itself does).

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/couch_sql_migration/couchsqlmigration.py (hunk 1 above)
Comment (millerdev) [reply corehq/apps/couch_sql_migration/couchsqlmigration.py]: This is a good suggestion. I do it in a future PR.

**Pull request merged**